### PR TITLE
Remove unused go libraries, ensure bazel works

### PR DESCRIPTION
**Pull request by fejta**
/assign @cjwagner @BenTheElder @ixdy 

```
bazel build //...
INFO: Analysed 818 targets (1 packages loaded).
INFO: Found 818 targets...
INFO: Elapsed time: 3.415s, Critical Path: 0.69s
INFO: Build completed successfully, 5 total actions

bazel test //...
INFO: Analysed 818 targets (1 packages loaded).
INFO: Found 720 targets and 98 test targets...
INFO: Elapsed time: 37.855s, Critical Path: 35.03s
INFO: Build completed successfully, 122 total actions
...
Executed 10 out of 98 tests: 98 tests pass.
```

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Is the script you used to remove the libraries something we want to check in? Will we need it again in the future?

**Comment by ixdy** [comment]
Won't these all get added back the next time somebody runs `dep`?

**Comment by fejta** [comment]
Yes I'll send the script in another PR. 

Jeff, once the script is in we can verify that it doesn't trim anything.

**Comment by chrislovecnm** [comment]
@fejta this busted my kops PR ... please see https://github.com/kubernetes/test-infra/issues/6173

**Comment by BenTheElder** [comment]
At least some of this will get added back next time someone runs `dep ensure` after adding a library. See https://github.com/kubernetes/test-infra/pull/6101#issuecomment-355795491

We could probably make this part of a verification presubmit and `hack/update-all.sh` to prevent that though.

File: prow/config.yaml [hunk 1]
```diff
@@ -5437,9 +5437,9 @@ presubmits:
         - "--service-account=/etc/service-account/service-account.json"
         - "--upload=gs://kubernetes-jenkins/pr-logs"
         - "--" # end bootstrap args, scenario args below
-        - "--build=//... -//vendor/..."
+        - "--build=//..."
         - "--install=gubernator/test_requirements.txt"
-        - "--test=//... //hack:verify-all -//vendor/..."
+        - "--test=//..."
```
Comment (BenTheElder) [inline prow/config.yaml]: doing this in a follow up, but there are a couple of other jobs that need this.

**Pull request merged**